### ISSUE TITLE
`catch->error` breaks with forms that are maps

### ISSUE THREAD

**Issue by rschmukler** [taoensso/telemere#29]
If you have map as your form you will get an exception that calling `catch->error` with a single map form is usually an error, even if you don't call it with a single map. Example:

```clojure
(tel/catch->error!
  {:id   :an-error
   :msg  "An error occurred"
   :data {:a 1
          :b 2}}
  {:foo 5})
;; => Invalid catch->error!
```

As a work around you can do

```clojure
(tel/catch->error!
  {:id   :an-error
   :msg  "An error occurred"
   :data {:a 1
          :b 2}}
  (do {:foo 5}))
```

**Comment by ptaoussanis** [comment]
@rschmukler Hi Ryan- sorry about that! This is a known issue that's been [fixed on master](https://github.com/taoensso/telemere/commit/93820f5c4ba66d4dc575f346172186b31a263401) and is just waiting on a new release.

I'm expecting to cut RC1 before the end of October, but your workaround looks good in the meantime 👍

**Comment by rschmukler** [comment]
Sorry for the double report, I should have searched a bit before reporting!

**Comment by ptaoussanis** [comment]
No worries! Actually, the problem was never reported - so wouldn't have been found in an issue. Just noticed it myself one day.

I'll keep this until there's an official release out with the fix 👍

**Comment by ptaoussanis** [comment]
Fixed in [v1.0.0-RC1](https://github.com/taoensso/telemere/releases/tag/v1.0.0-RC1).
